### PR TITLE
DPC-3839 add ability to add new org

### PR DESCRIPTION
**Pull request by Miki543**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3839

## 🛠 Changes

- new organization component
- new ToS component
- new success organization component
- edit organization controller to add new methods for backend
- tests for new components

## ℹ️ Context for reviewers

(Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.)

## ✅ Acceptance Validation
Screenshot 2024-03-08 at 9 56 59 AM
Screenshot 2024-03-08 at 9 58 31 AM
Screenshot 2024-03-08 at 9 58 55 AM
Screenshot 2024-03-08 at 9 58 55 AM

Screenshot 2024-03-08 at 9 59 38 AM

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: dpc-portal/app/components/page/organization/new_organization_component.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Page
+  module Organization
+    # Render a USWDS-styled new organization form for a controller.
+    class NewOrganizationComponent < ViewComponent::Base
+      attr_reader :npi_error
+
+      def initialize(npi_error)
```
Comment (jdettmannnava) [inline dpc-portal/app/components/page/organization/new_organization_component.rb:9]: `npi_error = ''` will make it optional, so Page::Organization::NewOrganizationComponent.new would be all you need unless there is an error.

File: dpc-portal/app/controllers/organizations_controller.rb [hunk 2]
```diff
@@ -21,6 +21,28 @@ def show
...
+      render(Page::Organization::NewOrganizationComponent.new(npi_error), status: :bad_request)
+    end
+  end
+
+  def npi_error
+    if params[:npi].blank?
+      "can't be blank"
+    elsif params[:npi].length != 10
+      'length has to be 10'
+    else
```
Comment (jdettmannnava) [inline dpc-portal/app/controllers/organizations_controller.rb:41]: Unnecessary; will return false without it.

**Review by MEspositoE14s — APPROVED** [review]
I'm far from a Ruby expert, but everything seems good to me, the tests pass, and the components render in Lookbook.

File: dpc-portal/app/components/page/organization/new_organization_success_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    and managing your access to the DPC API.
+  </p>
+  <p>Learn more about the Credential Delegate role.</p>
+  <div class="margin-top-5">
+    <div class="display-flex flex-row flex-start" style="gap:20px;">
+        <div class="flex-align-self-center">
+            <%= render Core::Button::ButtonComponent.new(label: 'Assign CD now', destination: root_path)%>
+        </div>
+        <div class="flex-align-self-center">
+            <%= link_to 'Assign CD later', root_path %>
```
[the remark this answers is not in the document]
Comment (jdettmannnava) [reply dpc-portal/app/components/page/organization/new_organization_success_component.html.erb:22]: Just the spec

File: dpc-portal/app/components/page/organization/tos_form_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  <p>To manage this organization's credentials, you must agree to the DPC API's Terms of Service.</p>
+  <div style="width: 100%; height: 200px; background-color: silver" class="margin-bottom-5">&nbsp;</div>
+    <div>
+        <ul class="usa-button-group">
+          <li class="usa-button-group__item">
+            <%= button_to 'I have read and accepted the Terms of Service', sign_tos_organization_path('place-holder'), class: 'usa-button' %>
+          </li>
+          <li class="usa-button-group__item">
+            <button type="button" class="usa-button usa-button--unstyled padding-105 text-center" data-close-modal>
+              <%= link_to 'Cancel', organizations_path %>
```
[the remark this answers is not in the document]
Comment (jdettmannnava) [reply dpc-portal/app/components/page/organization/tos_form_component.html.erb:15]: Just the spec

File: dpc-portal/config/routes.rb [hunk 5]
```diff
@@ -19,17 +19,20 @@
...
       get 'success', on: :member
     end
+    get 'tos_form', on: :member
+    post 'sign_tos', on: :member
+    get 'success', on: :member
   end
 
   match '/download_snippet', to: 'public_keys#download_snippet', as: 'download_snippet', via: :post
-
+\
```
Comment (ashley-weaver) [inline dpc-portal/config/routes.rb:35]: stray character?

**Review by kyeah — APPROVED** [review]
👍 Having some trouble with my local environment, but from reviewing the PR itself it looks good.

**Pull request merged**